src/components/CaptureTable.js

```diff
@@ -113,6 +113,8 @@ const CaptureTable = () => {
   const scrollRef = createRef();
   const classes = useStyle();
 
+  const [page, setPage] = useState(capturesContext.page);
```

Comment:
There's no need to store the `page` as local state – `CapturesContext` maintains this value for us, and we want to avoid duplicating it.
Were you trying to address the lag between clicking `>` and the page number changing, and that clicking multiple times doesn't result in the same number of page changes?
If so, I'd suggest looking at `CapturesContext:getCapturesAsync()` and moving the state updates before the API call:
```
    setPage(page);
    setRowsPerPage(rowsPerPage);
    setOrderBy(orderBy);
    setOrder(order);
    setFilter(filter);
```

Reply:
Previously, the `handlePageChange()` function looks like this:

```js
const handlePageChange = () => {
    loadCaptures({
      page: capturesContext.page + 1, // the location where the bug happens
      rowsPerPage: capturesContext.rowsPerPage,
      filter: capturesContext.filter,
    });
  };
```

This caused the page to moving forward even when clicking `<`, while clicking `<` supposed to move us back to the previous page. When checking the [TablePagination API](https://mui.com/api/table-pagination/#props), I notice that `handlePageChange()` accepts 2 arguments: a React mouse event and the new page value. As such, I thought that I need to somehow change the `page` state into the new page value before calling `loadCaptures()`, which explains why I create a local `page` state. I'll try to find a way to make use of the `CapturesContext:getCapturesAsync()` per your suggestion, but I'm not sure how I can do it.

Reply:
You've fixed the problem well, you just don't need to set a local state for `page`. Just use `capturesContext.page` directly.

When you pass `page: newPage` in `loadCaptures()`, `capturesContext.page` gets updated.